src/contextkit-subscriber/property.cpp

```diff
@@ -189,7 +189,9 @@ class UnsubscribeRequest : public Event
         , key_(key)
         , done_(std::move(done))
     {}
-    virtual ~UnsubscribeRequest() {}
+    virtual ~UnsubscribeRequest() {
+        done_.set_value();
+    }
 
     UnsubscribeRequest(SubscribeRequest const&) = delete;
```

Comment:
Ah, just deletion of copy ctor, before devs put it in the ```private``` section but it was still possible to use it and intention was not obvious.